Python/pystate.c

```diff
@@ -1066,6 +1064,11 @@ _PyThreadState_Swap(struct _gilstate_runtime_state *gilstate, PyThreadState *new
 #endif
 
     _PyRuntimeGILState_SetThreadState(gilstate, newts);
+#ifdef _Py_THREAD_LOCAL
+    if (newts != NULL) {
+        _py_current_interpreter = newts->interp;
+    }
+#endif
```

Comment:
It feels a bit strange that this is the only place we need to set this. 🙂

Reply:
(To be clear, this is only an observation on how unexpected it is that there is only one place where the thread-local needs to be updated.  Nothing needs to be done about it. 😄)